src/components/auth/Register.jsx

```diff
@@ -1,9 +1,332 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import Modal from 'react-modal';
+import { registerStudent, registerTeacher } from '../../util/DataBaseRequests';
+import FormInput from '../common/FormInput';
+import FormSubmitBtn from '../common/FormSubmitBtn';
+import FormErrorMsg from '../common/FormErrorMsg';
+
 const Register = () => {
+  const [userRole, setUserRole] = useState('student');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [dateOfBirth, setDateOfBirth] = useState('');
+  const [isAdultNameRequired, setIsAdultNameRequired] = useState(false);
+  const [adultName, setAdultName] = useState('');
+  const [adultNameError, setAdultNameError] = useState('');
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [confirmEmail, setConfirmEmail] = useState('');
+  const [confirmEmailError, setConfirmEmailError] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const navigate = useNavigate();
+
+
+  const handleUserRegistration = async (event) => {
+    event.preventDefault();
+
+    let isDataValid = true;
+
+    if (email.trim() === '') {
```

Comment:
Form validation can be its own component.

Reply:
I moved form validation logic was moved to a separate function. But this function is still inside the Register component. Sorry, I do not understand how to make validation a separate component. Could you please explain further what you meant by 'can be its own component'?

Reply:
I moved form rendering into a separate component RegisterForm

Reply:
That works. Ideally you would have the form in its own component and in that component you could run validation logic whereas `Register` would handle that actual request to the API. But this works for now and we can make that a stretch goal if we have to for refactoring.